### PR TITLE
add Feature 

### PR DESCRIPTION
**Pull request by BaronMsk**
add feature kube_proxy_enable
add feature maintenance_new_nodes
add feature containerd_sandbox_image

### CODE REVIEW

**Comment by ArsenyBelorukov** [comment]
I suggest instead of maintenance_new_nodes use something like node_registation_tains (Array) to have the ability to use other taints too.

**Comment by ArsenyBelorukov** [comment]
@BaronMsk  I don't know about tests, sorry.
I am just in the adoption process of this module.

I have another question about your PR:
you use `kube_proxy_enable` not to add `kubeproxy.config` to kubeadm config, but do not set the skipping the phase addon/kube-proxy.  How should it work?

**Comment by BaronMsk** [comment]
kube-proxy is not a phase, it's an addon. kubeadm will only show a warning and continue ```Notice: /Stage[main]/Kubernetes::Cluster_roles/Kubernetes::Kubeadm_join[k8s-test-worker-dtln-3]/Exec[kubeadm join]/returns: W1027 14:23:31.375960    3628 configset.go:78] Warning: No kubeproxy.config.k8s.io/v1alpha1 config is loaded. Continuing without it: configmaps "kube-proxy" is forbidden: User "system:bootstrap:e4247f" cannot get resource "configmaps" in API group "" in the namespace "kube-system"```

**Comment by ArsenyBelorukov** [comment]
@BaronMsk 
Deploy kube-proxy is performed in the addon/kube-proxy phase.
Without `kubeproxy.config` it is deployed anyway:
```
# grep kind /etc/kubernetes/config.yaml
kind: InitConfiguration
kind: ClusterConfiguration
# kubeadm init --config '/etc/kubernetes/config.yaml'
...
[addons] Applied essential addon: kube-proxy
...
# kubectl -n kube-system get daemonsets
NAME         DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR            AGE
kube-proxy   1         1         1       1            1           kubernetes.io/os=linux   60s
```

**Comment by BaronMsk** [comment]
yes I understand what you mean... I think kubeadm doesn't allow to disable addons ((((.  The problem is that if do not remove kube-proxy from config, then when updating kubeadm, it will deploy it again

**Comment by GSPatton** [comment]
Hi @BaronMsk . Thank you for your contribution. In order for us to best address your PR, we'd appreciate if you could add a more descriptive title and a more detailed description outlining the context of the problem you're addressing and the steps you took to make the necessary changes. 

We have a `CONTRIBUTION.md` in each module repo linking to [this](https://puppetlabs.github.io/iac/docs/contributing_to_a_module.html) which outlines instructions on how to best contribute to Puppet modules. 

Thanks

File: templates/v1beta3/config_worker.yaml.erb [hunk 1]
```diff
@@ -14,9 +14,16 @@ discovery:
     - 'sha256:<%= @discovery_token_hash %>'
 nodeRegistration:
   name: <%= @node_name %>
+<%- if @node_extra_taints -%>
+  taints:
+  <%- @node_extra_taints.each do |item| -%>
+  - key: <%= item['key'] %>
+    value: <%= item['value'] %>
```
Comment (ArsenyBelorukov) [inline templates/v1beta3/config_worker.yaml.erb:21]: value is not obligatory,
what do you think about something like 
```
    <%- if item['value'] -%>
    value: <%= item['value'] %>
    <%- end -%>
```

Comment (ArsenyBelorukov) [reply templates/v1beta3/config_worker.yaml.erb:21]: What about adding extra taints to the config_kubeadm.yaml.erb ?

**Comment by BaronMsk** [comment]
I am closing this PR. I will continue to work in other PRs

**Pull request closed without merging**